routers/web/repo/issue.go

```diff
@@ -1637,6 +1638,27 @@ func ViewIssue(ctx *context.Context) {
 	ctx.Data["HasProjectsWritePermission"] = ctx.Repo.CanWrite(unit.TypeProjects)
 	ctx.Data["IsRepoAdmin"] = ctx.IsSigned && (ctx.Repo.IsAdmin() || ctx.User.IsAdmin)
 	ctx.Data["LockReasons"] = setting.Repository.Issue.LockReasons
+	var hiddenEvents *forms.UpdateCommentTypeForm
+	if ctx.IsSigned {
+		eventsRaw, err := user_model.GetSettings(ctx.User.ID, []string{"hidden_comment_types"})
+		if err != nil {
+			ctx.ServerError("GetSettings", err)
+			return
+		}
+		if eventsRaw["hidden_comment_types"] != nil && eventsRaw["hidden_comment_types"].SettingValue != "" {
+			err = json.Unmarshal([]byte(eventsRaw["hidden_comment_types"].SettingValue), &hiddenEvents)
+			if err != nil {
+				ctx.ServerError("json.Unmarshal", err)
```

Comment:
Here and above, for non-critical errors, I think we can just output an error log, and continue to render the page. 

500 Internal Error is too annoying for users